### ISSUE TITLE
no licensing information

### ISSUE THREAD

**Issue by stringertheory** [pygrace/pygrace#6]
MIT? BSD?

**Comment by deanmalmgren** [comment]
my vote (if i even get one) is MIT

**Comment by stouffer** [comment]
I would vote for MIT as well but I'll have to figure out how compatible it is with the license of McKern's pygrace which I think is three-clause BSD. Do either of you know if MIT can cope with having something using BSD underneath its umbrella?

**Comment by deanmalmgren** [comment]
I don't remember, but is McKern's pygrace included in this package? If we
don't _literally_ distribute his source code anywhere in PyGrace, then we
can release our project under anything we want

**Comment by stouffer** [comment]
Not yet it isn't, but if you look at the altmerge branch it's there and in hibernation for the moment that I get a green light from him to do so. It was our plan however many years ago he and I talked about it, but we ended up never finding a chance to touch base and test things thoroughly enough to risk the users of his code being inconvenienced by a buggy implementation.

**Comment by deanmalmgren** [comment]
ok, well then why don't we go with BSD since that appears to be PyGrace's
destiny anyway

**Comment by mmckerns** [comment]
I'd recommend BSD… although MIT is basically the same as BSD, as far as anything that really matters.  You'll want to be able to commercialize a fork if it, and as consultants to enable a business to use and fork your code.

If you want to ever resurrect the altmerge attempt… I'm fine with that happening.  That's the primary way you'd get to put PyGrace on pypi (#2).

**Comment by utf** [comment]
Is there any update on this?

It would be good to have pygrace on PyPi

**Comment by mmckerns** [comment]
It is on pypi... sort of.  https://pypi.python.org/pypi/pygrace/0.4

`pygrace` is BSD and has been for over 10 years, and I think that was the conclusion.

@utf: Note that there is the original `pygrace` and a second orthogonal codebase (which has never been merged with the original and appears) in the master of this git repository.  For the code that I think you are probably interested in (the one in the master of this git repository), the code would need to be merged with the original `pygrace` for it to get onto pypi.  Or, it could go on under a different name, but that's kind of confusing.

I've thought recently about resurrecting the push to merge the two codes, but it hasn't made it off of the back-burner.

**Comment by utf** [comment]
@mmckerns: I stumbled on the pypi `pygrace` version earlier. It seems that the approach and functionality of the `pygrace` on pypi is quite different to the functionality of the pypi in this repository. I agree that putting this version up under a different name may be a little confusing, however, merging these two codes does seem like rather a lot of hassle and I know at least several people that would appreciate having access to this repository on pypi.

I notice there is an altmerge branch in this repository which has been configured for setuptools. Would you consider pushing this code to pypi under a different name, say `gracepy`?

**Comment by stouffer** [comment]
Yes, the final verdict was BSD since the plan was and is still to incorporate @mmckerns' version into this code. I'll attend to adding the license asap (which might even mean today).

@mmckerns: I went back and looked through some old emails. It seems that as of November 2013 (!) I was convinced that the altmerge branch would be fully backwards compatible with the `pygrace` on pypi, and what I mostly needed was for you to look over the code to make sure you agreed. The last thing I wanted, or still want, is for us merging these two packages together to screw up users of your side of things, especially since you're no longer actively maintaining it. Any chance you could pick this back up and see if altmerge gets your green light and then I can aim to update the version on pypi?

**Comment by mmckerns** [comment]
@stouffer: I remember the code base being pretty orthogonal, and that I did the first merge attempt... so I shouldn't have issues with it unless something had been changed.  If my memory serves me, there have been some name-spacing issues that may have been the eventual hang-up.  

FYI, I'm no longer actively maintaining the code because It's a fairly complete solution for the portion of the language bindings it covers.  Unless there are new language features added to Grace, there's not much to change aside from any reported bugs. It's more so that I don't intend to expand the feature set it currently covers.

Anyway, in short, yes, I'll look into it, and let's see what we can do to get the codes merged happily and everything onto pypi.

**Comment by gnibeil** [comment]
Hi,
I'm sorry I missed this discussion here. Since I found here
https://sourceforge.net/projects/pygrace/
and here
https://pypi.org/project/pygrace/
something about modified BSD I put my fork to BSD-3 clause if somebody has a problem with that let me know.

**Comment by mmckerns** [comment]
It's BSD
